graph/Guidelines.md

```diff
@@ -0,0 +1,79 @@
+# Microsoft Graph REST API Guidelines
+
+## History
+
+| Date        | Version | Notes                                               |
+| ----------- | ------- | --------------------------------------------------- |
+| 2021-Mar-31 | v3.3    | 1st public release of the Graph REST API Guidelines |
+
+## Introduction
+
+The Microsoft Graph REST API guidelines are an extension of the [Microsoft REST API guidelines][1]. Readers of this document are assumed to be also reading the [Microsoft REST API guidelines][1] and be familiar with them. Microsoft Graph guidance is a superset of the Microsoft API guidelines and services should follow them *except* where this document outlines specific differences or exceptions to those guidelines. This document does contain additional Microsoft Graph specific guidance and additional details.
+
+## Dictionary types
+
+Dictionaries, sometimes called maps, are a collection of name-value pairs. They allow dynamic data sets to be accessed in a systematic maner and are a good compromise between a stricly defined ahead of time structure with all its named properties and between a loosely defined dynamic object (i.e. OData OpenTypes).
+
+More information:
+
+- [OData reference](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Core.V1.md#dictionary)
+
+### When to use dictionary types
+
+Before using a dictionary type in your API definition you should make sure your scenario fits the following criteria:
+
+- The data set is dynamic and you don't know in advance how many entries could be part of that data set.
+- The data values are related to one another semanticaly.
+- The data values do not need to be individually addressable for CRUD operations.
```

Comment:
Why does this restriction exist?  It should be perfectly valid to address an individual element from the dictionary -- i.e., myEntity/AdditionalProperties/EyeColor

Reply:
Shouldn't you use an entity set then if you want things to be individually addressable?

Reply:
If I have a schematized collection then yes, I would use an entityset/collection-valued navigation property.

---
In reply to: [601332605](https://github.com/microsoft/api-guidelines/pull/226#discussion_r601332605) [](ancestors = 601332605)

Reply:
addressed on the call, it is possible and the sentence has been removed. resolving.